pkg/clients/rds/rds.go

```diff
@@ -0,0 +1,190 @@
+// Client defines RDS client operations
+type Client interface {
+	DescribeDBInstance(id string) (*DBInstance, error)
+	CreateDBInstance(*CreateDBInstanceRequest) (*DBInstance, error)
+	DeleteDBInstance(id string) error
+}
+
+// DBInstance defines the DB instance information
+type DBInstance struct {
+	// Database engine
+	Engine string
+
+	// Instance status
+	Status string
+
+	// Endpoint specifies the connection endpoint.
+	Endpoint *v1alpha1.Endpoint
+}
+
+// CreateDBInstanceRequest defines the request info to create DB Instance
+type CreateDBInstanceRequest struct {
+	Name                  string
+	Engine                string
+	EngineVersion         string
+	SecurityIPList        string
+	DBInstanceClass       string
+	DBInstanceStorageInGB int
+	Username              string
+	Password              string
+}
+
+type client struct {
+	rdsCli *alirds.Client
+}
+
+// NewClient creates new RDS RDSClient
+func NewClient(ctx context.Context, accessKeyID, accessSecret, region string) (Client, error) {
+	rdsCli, err := alirds.NewClientWithAccessKey(region, accessKeyID, accessSecret)
+	if err != nil {
+		return nil, err
+	}
+	c := &client{rdsCli: rdsCli}
+	return c, nil
+}
+
+func (c *client) DescribeDBInstance(id string) (*DBInstance, error) {
+	request := alirds.CreateDescribeDBInstancesRequest()
+	request.Scheme = "https"
+
+	request.DBInstanceId = id
+
+	response, err := c.rdsCli.DescribeDBInstances(request)
+	if err != nil {
+		return nil, err
+	}
+	if len(response.Items.DBInstance) == 0 {
+		return nil, ErrDBInstanceNotFound
+	}
+	rsp := response.Items.DBInstance[0]
+	in := &DBInstance{
+		Engine: rsp.Engine,
+		Status: rsp.DBInstanceStatus,
+	}
+
+	return in, nil
+}
+
+func (c *client) CreateDBInstance(req *CreateDBInstanceRequest) (*DBInstance, error) {
+	request := alirds.CreateCreateDBInstanceRequest()
+	request.Scheme = "https"
+
+	request.DBInstanceDescription = req.Name
+	request.Engine = req.Engine
+	request.EngineVersion = req.EngineVersion
+	request.DBInstanceClass = req.DBInstanceClass
+	request.DBInstanceStorage = requests.NewInteger(req.DBInstanceStorageInGB)
+	request.SecurityIPList = req.SecurityIPList
+	request.DBInstanceNetType = "Internet"
+	request.PayType = "Postpaid"
+
+	resp, err := c.rdsCli.CreateDBInstance(request)
+	if err != nil {
+		return nil, err
+	}
+
+	accReq := alirds.CreateCreateAccountRequest()
+	accReq.Scheme = "https"
+	accReq.DBInstanceId = resp.DBInstanceId
+	accReq.AccountName = req.Username
+	accReq.AccountPassword = req.Password
+
+	_, err = c.rdsCli.CreateAccount(accReq)
+	if err != nil {
+		return nil, err
+	}
+
+	return &DBInstance{
+		Endpoint: &v1alpha1.Endpoint{
+			Address: resp.ConnectionString,
+			Port:    resp.Port,
+		},
+	}, nil
+}
+
+func (c *client) DeleteDBInstance(id string) error {
+	request := alirds.CreateDeleteDBInstanceRequest()
+	request.Scheme = "https"
+
+	request.DBInstanceId = id
+
+	_, err := c.rdsCli.DeleteDBInstance(request)
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
```suggestion
	return err
```

Nit: this logic is superfluous.